### PR TITLE
Added support for onResponseHooks - changing version to 1.3

### PR DESCRIPTION
**Pull request by deepeshhmehta**
* split up onRequest and onResponse types
* improved the type definitions
* copied over default onResponseHook from fast-gateway to use as the streamer / last response
* added same functionality as onRequest with break the flow using booleans

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,4 +13,62 @@ const fgMultipleHooks = async (req, res, ...multipleHooks) => {
...
+    } else {
+        res.statusCode = stream.statusCode
+        pump(stream, res)
+    }
+}
+
+const onResponseHooks = async (req, res, stream, last, ...multipleHooks) => {
+    let shouldAbort = false;
+    const promises = multipleHooks.map(async oneHook => {
+        shouldAbort = shouldAbort || await oneHook(req, res);
```
Comment (hamilton-lima) [inline index.js:54]: every hook should receive the the stream

File: index.js [hunk 2]
```diff
@@ -7,4 +13,62 @@ const fgMultipleHooks = async (req, res, ...multipleHooks) => {
...
+            // remove transfer-encoding header
+            const transferEncoding = stream.headers[TRANSFER_ENCODING_HEADER_NAME].replace(/(,( )?)?chunked/, '')
+            if (transferEncoding) {
+                // header format includes many encodings, example: gzip, chunked
+                res.setHeader(TRANSFER_ENCODING_HEADER_NAME, transferEncoding)
+            } else {
+                res.removeHeader(TRANSFER_ENCODING_HEADER_NAME)
+            }
+
+            if (!stream.headers['content-length']) {
```
Comment (hamilton-lima) [inline index.js:32]: what about only this block as the implementation for this hook?

File: index.js [hunk 3]
```diff
@@ -7,4 +13,62 @@ const fgMultipleHooks = async (req, res, ...multipleHooks) => {
     return shouldAbort;
 };
 
-module.exports = fgMultipleHooks;
\ No newline at end of file
+const defaultLastOnResponse = async (req, res, stream) => {
```
Comment (hamilton-lima) [inline index.js:16]: the name "default" does not give a clear indication of what this hook does, maybe afterResponseSendStreamUntouched or something less verbose but with more meaning

File: index.js (hunk 2 above)
Comment (deepeshhmehta) [reply index.js:32]: I would like the consumer to have an option to create a hook with no stream responsibility. having the whole functionality in a default setup allows me to do that

File: index.js (hunk 1 above)
Comment (deepeshhmehta) [reply index.js:54]: that is particularly what i have been avoiding, any operation on the stream can and should be done in the last hook.

Comment (deepeshhmehta) [reply index.js:54]: on second thoughts, it is still the developer's consumer's responsibility, I'll do the need full changes to allow stream in every Hook

File: index.js (hunk 2 above)
Comment (deepeshhmehta) [reply index.js:38]: have implemented a different style, i think it covers this flexibility

**Pull request merged**